classes/PrestaShopCollection.php

```diff
@@ -401,6 +401,21 @@ public function getFirst()
         return $this[0];
     }
 
+    /**
+     * Retrieve the last result.
+     *
+     * @return ObjectModel|false
+     */
+    public function getLast()
+    {
+        $this->getAll();
+        if (!count($this)) {
+            return false;
+        }
+
+        return $this[count($this) - 1];
```

Comment:
```suggestion
        return end($this);
```

Reply:
MM, I'm not sure 
I don't want to move the array cursor

Reply:
But how can you be sure that the key of the array is well `count($this) - 1` ?

Reply:
Because that's how this class is supposed to behave, it stores db results in `$this->results` which is a regular array (I mean not associative). Besides if you look at the `getFirst` function it uses `return $this[0];` and all the `getOffset` `setOffset` also use an index to access the results

Reply:
:ok_hand: